web/pdf_find_controller.js

```diff
@@ -41,447 +42,440 @@ var CHARACTERS_TO_NORMALIZE = {
-      this.resumePageIdx = null;
-      this.state = null;
-      this.dirtyMatch = false;
-      this.findTimeout = null;
+  reset() {
+    this.startedTextExtraction = false;
+    this.extractTextPromises = [];
+    this.pendingFindMatches = Object.create(null);
+    this.active = false; // If active, find results will be highlighted.
+    this.pageContents = []; // Stores the text for each page.
+    this.pageMatches = [];
+    this.pageMatchesLength = null;
+    this.matchCount = 0;
+    this.selected = { // Currently selected match.
+      pageIdx: -1,
+      matchIdx: -1,
+    };
+    this.offset = { // Where the find algorithm currently is in the document.
+      pageIdx: null,
+      matchIdx: null,
+    };
+    this.pagesToSearch = null;
+    this.resumePageIdx = null;
+    this.state = null;
+    this.dirtyMatch = false;
+    this.findTimeout = null;
+
+    this._firstPagePromise = new Promise((resolve) => {
+      this.resolveFirstPage = resolve;
+    });
+  }
 
-      this._firstPagePromise = new Promise((resolve) => {
-        this.resolveFirstPage = resolve;
-      });
-    },
+  normalize(text) {
+    return text.replace(this.normalizationRegex, function (ch) {
+      return CHARACTERS_TO_NORMALIZE[ch];
+    });
+  }
 
-    normalize: function PDFFindController_normalize(text) {
-      return text.replace(this.normalizationRegex, function (ch) {
-        return CHARACTERS_TO_NORMALIZE[ch];
-      });
-    },
-
-    // Helper for multiple search - fills matchesWithLength array
-    // and takes into account cases when one search term
-    // include another search term (for example, "tamed tame" or "this is").
-    // Looking for intersecting terms in the 'matches' and
-    // leave elements with a longer match-length.
-
-    _prepareMatches: function PDFFindController_prepareMatches(
-        matchesWithLength, matches, matchesLength) {
-
-      function isSubTerm(matchesWithLength, currentIndex) {
-        var currentElem, prevElem, nextElem;
-        currentElem = matchesWithLength[currentIndex];
-        nextElem = matchesWithLength[currentIndex + 1];
-        // checking for cases like "TAMEd TAME"
-        if (currentIndex < matchesWithLength.length - 1 &&
-            currentElem.match === nextElem.match) {
-          currentElem.skipped = true;
-          return true;
-        }
-        // checking for cases like "thIS IS"
-        for (var i = currentIndex - 1; i >= 0; i--) {
-          prevElem = matchesWithLength[i];
-          if (prevElem.skipped) {
-            continue;
-          }
-          if (prevElem.match + prevElem.matchLength < currentElem.match) {
-            break;
-          }
-          if (prevElem.match + prevElem.matchLength >=
-              currentElem.match + currentElem.matchLength) {
-            currentElem.skipped = true;
-            return true;
-          }
-        }
-        return false;
+  // Helper for multi-term search that fills the `matchesWithLength` array
```

Comment:
Nit: Thanks for improving the comment, but shouldn't it use JSDoc formatting since it's documenting a method?